recipes/lzham/all/conanfile.py

```diff
@@ -0,0 +1,221 @@
+import os
+
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import (
+    apply_conandata_patches,
+    copy,
+    export_conandata_patches,
+    get,
+    replace_in_file,
+    rmdir
+)
+from conan.tools.microsoft import (
+    MSBuild, MSBuildDeps, MSBuildToolchain, VCVars, is_msvc, vs_layout
+)
+
+required_conan_version = ">=1.52.0"
+
+SLN_FILE = "lzham.sln"
+
+
+class PackageConan(ConanFile):
+    name = "lzham"
+
+    description = (
+        "Compression algorithm similar compression ratio and faster "
+        "decompression than LZMA."
+    )
+
+    license = "LicenseRef-LICENSE"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/richgel999/lzham_codec"
+    topics = ("compression", "lz-compression")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+
+        if not is_msvc(self):
+            # Remove lzhamtest from root CMakeLists.txt.
```

Comment:
This should be a patch (or maybe contribute to upstream with an option to build tests 😉 )